src/mod/asr_tts/mod_unimrcp/mod_unimrcp.c

```diff
@@ -2460,6 +2475,8 @@ static switch_status_t recog_channel_load_grammar(speech_channel_t *schannel, co
 		}
 		if (schannel->state != SPEECH_CHANNEL_READY) {
 			status = SWITCH_STATUS_FALSE;
+			/* set mrcp channel object to NULL */
+			schannel = (speech_channel_t *) mrcp_application_channel_object_set(schannel->unimrcp_channel, NULL);
```

Comment:
No need to reset the channel object to NULL here since schannel is still alive. If we return a false here it will still call recog_asr_close which we can set the schannel to NULL in the mrcp channel.